File: algorithms/interview-preparation-kit-2/php/code-response/src/HackerrankInString/HackerrankInString.php
<?php

namespace HackerrankInString;

class HackerrankInString
{
  public static function check($s)
  {
    $hackerrank = 'hackerrank';
    for ($i = 0, $index = 0; $i < strlen($s); $i++) {
      if (\substr($s, $i, 1) === \substr($hackerrank, $index, 1)) {
        $index++;
      }
      if ($index == strlen($hackerrank)) {
        return 'YES';
      }
    }
    return 'NO';
  }
}

File: algorithms/interview-preparation-kit-2/ruby/code-response/hackerrankInString/hackerrankInString.rb
def hackerrankInString(s)
  hackerrank = 'hackerrank';
  index = 0
  for i in 0...s.length do
    if (s[i] == hackerrank[index])
      index = index+1
    end
    if (index == hackerrank.length)
      return 'YES';
    end
  end
  'NO'
end
